l1-contracts-foundry/script/DeployL1.s.sol

```diff
@@ -0,0 +1,703 @@
+            "l1.bridges",
+            "shared_bridge_proxy_addr",
+            addresses.bridges.sharedBridgeProxy
+        );
+
+        vm.serializeAddress("l1.config", "governor_addr", config.contracts.governorAddress);
+        vm.serializeUint(
+            "l1.config",
+            "diamond_init_batch_overhead_l1_gas",
+            config.contracts.diamondInitBatchOverheadL1Gas
+        );
+        vm.serializeUint(
+            "l1.config",
+            "diamond_init_max_pubdata_per_batch",
+            config.contracts.diamondInitMaxPubdataPerBatch
+        );
+        vm.serializeUint(
+            "l1.config",
+            "diamond_init_max_l2_gas_per_batch",
+            config.contracts.diamondInitMaxL2GasPerBatch
+        );
+        vm.serializeUint(
+            "l1.config",
+            "diamond_init_priority_tx_max_pubdata",
+            config.contracts.diamondInitPriorityTxMaxPubdata
+        );
+        vm.serializeUint(
+            "l1.config",
+            "diamond_init_minimal_l2_gas_price",
+            config.contracts.diamondInitMinimalL2GasPrice
+        );
+        vm.serializeBytes32("l1.config", "recursion_node_level_vk_hash", config.contracts.recursionNodeLevelVkHash);
+        vm.serializeBytes32("l1.config", "recursion_leaf_level_vk_hash", config.contracts.recursionLeafLevelVkHash);
+        vm.serializeBytes32(
+            "l1.config",
+            "recursion_circuits_set_vks_hash",
+            config.contracts.recursionCircuitsSetVksHash
+        );
+        string memory l1Config = vm.serializeUint(
+            "l1.config",
+            "priority_tx_max_gas_limit",
+            config.contracts.priorityTxMaxGasLimit
+        );
+
+        vm.serializeUint("l1", "chain_id", config.chainId);
+        vm.serializeUint("l1", "era_chain_id", config.eraChainId);
+        vm.serializeString("l1", "bridgehub", l1Bridgehub);
+        vm.serializeString("l1", "state_transition", l1StateTransition);
+        vm.serializeString("l1", "config", l1Config);
+        vm.serializeAddress("l1", "deployer_addr", config.deployerAddress);
+        string memory l1 = vm.serializeString("l1", "bridges", l1Bridges);
+
+        string memory toml = vm.serializeString("toml", "l1", l1);
+        string memory root = vm.projectRoot();
+        string memory path = string.concat(root, "/script-out/output-deploy-l1.toml");
+        vm.writeToml(toml, path);
+    }
+
+    function deployViaCreate2(bytes memory _bytecode) internal returns (address) {
+        if (_bytecode.length == 0) {
+            revert("Bytecode is not set");
+        }
+        address contractAddress = vm.computeCreate2Address(
+            config.contracts.create2FactorySalt,
+            keccak256(_bytecode),
+            addresses.create2Factory
+        );
+        if (contractAddress.code.length != 0) {
+            return contractAddress;
+        }
+
+        vm.broadcast();
+        (bool success, bytes memory data) = addresses.create2Factory.call(
```

Comment:
not sure what create2 factory here is used, the one which we use on mainnet has `deploy` function in it

Reply:
Create2 factory being used here is the deterministic deployment proxy https://github.com/Arachnid/deterministic-deployment-proxy